python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/resource.py

```diff
@@ -100,6 +109,30 @@ def extract_resource_metadata(
             for job in load_package.get("jobs", [])
             if job.get("table_name") == resource.table_name
         ]
+        rows_loaded = dlt_pipeline.last_trace.last_normalize_info.row_counts.get(
+            str(resource.table_name), None
+        )
+        if rows_loaded:
+            base_metadata["rows_loaded"] = MetadataValue.int(rows_loaded)
+        try:
+            with dlt_pipeline.sql_client() as client:
+                with client.execute_query(
+                    f"""
+                        SELECT
+                        count(*) as row_count
+                        FROM
+                        {resource.table_name}
+                    """,
+                ) as cursor:
+                    row_count_result = cursor.fetchone()
```

Comment:
I think computing the full row count should be an opt-in behavior. While querying the count for small and moderately sized tables is fast and relatively low cost, I have concern that this could result in an undesired cost/compute for very large tables.

Reply:
That's fair. I guess it would be best to have a separate method called `fetch_row_counts()` such as `SlingResource.get_column_info_for_table()` and `DbtEventIterator.fetch_row_counts()`? Then users can opt-in through chaining: `dlt.run().fetch_row_count()`

Reply:
Yes that sounds great! Thank you.

Reply:
Looking a bit closer, I see that both the Sling and dbt resources use their own event iterators in order to chain metadata methods. I could either try to add a `DltEventIterator`, or add a `fetch_row_counts: Optional[bool] = False` flag to `DltResource.run()`. 

Adding an event iterator would increase the scope of this PR quite a bit but makes fetching metadata more consistent with the other resources. Do you still think I should proceed with implementing a `DltResource.fetch_row_count()` method?

Reply:
Perhaps we keep the `rows_loaded` metadata that is included in the `LoadInfo` in this PR, and move the `row_count` with a possible iterator to a follow-up PR?

Reply:
Agreed 👍🏻